drivers/net/dummy_rs.rs

```diff
@@ -0,0 +1,241 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Rust dummy network driver
+//!
+//! This is a demonstration of what a small driver looks like in Rust, based on drivers/net/dummy.c.
+//! This code is provided as a demonstration only, not as a proposal to mass-rewrite existing drivers in Rust
+//!
+//! The purpose of this driver is to provide a device to point a
+//! route through, but not to actually transmit packets.
+//!
+//! Why?  If you have a machine whose only connection is an occasional
+//! PPP/SLIP/PLIP link, you can only connect to your own hostname
+//! when the link is up.  Otherwise you have to use localhost.
+//! This isn't very consistent.
+//!
+//! One solution is to set up a dummy link using PPP/SLIP/PLIP,
+//! but this seems (to me) too much overhead for too little gain.
+//! This driver provides a small alternative. Thus you can do
+//!
+//! [when not running slip]
+//! 	ifconfig dummy slip.addr.ess.here up
+//! [to go to slip]
+//! 	ifconfig dummy down
+//! 	dip whatever
+//!
+//! This was written by looking at Donald Becker's skeleton driver
+//! and the loopback driver.  I then threw away anything that didn't
+//! apply!	Thanks to Alan Cox for the key clue on what to do with
+//! misguided packets.
```

Comment:
Hmm... this is a bit confusing now, it sounds like you are talking here since your name is right after the text (rather than coming from `dummy.c`, even if you reference it later).

Reply:
Oh, yes. Should reword that

Reply:
Reworded it like this:
~~~
//! This was written by looking at the dummy network driver from Nick
//! Holloway, which was written by looking at Donald Becker's skeleton driver
//! and the loopback driver